geo/src/algorithm/buffer.rs

```diff
@@ -0,0 +1,835 @@
+        };
+        check_buffer_test_case(&actual.into(), &expected_output_from_jts.into()).unwrap();
+    }
+
+    #[test]
+    fn buffer_linestring_empty() {
+        let line_string = wkt! { LINESTRING EMPTY };
+        let actual = line_string.buffer(2.0);
+        assert_eq!(actual.0.len(), 0);
+    }
+
+    #[test]
+    fn buffer_linestring_length_1() {
+        let line_string = wkt! { LINESTRING(2.0 3.0,2.0 3.0) };
+
+        let actual = line_string.buffer(2.0);
+        let expected_output_from_jts = wkt! {
+            POLYGON ((4. 3., 3.961570560806461 2.6098193559677436, 3.8477590650225735 2.2346331352698203, 3.6629392246050907 1.8888595339607956, 3.414213562373095 1.585786437626905, 3.1111404660392044 1.3370607753949095, 2.7653668647301797 1.1522409349774265, 2.390180644032257 1.0384294391935391, 2. 1., 1.6098193559677436 1.0384294391935391, 1.2346331352698205 1.1522409349774265, 0.8888595339607961 1.3370607753949093, 0.5857864376269051 1.5857864376269049, 0.3370607753949093 1.8888595339607956, 0.1522409349774265 2.2346331352698203, 0.0384294391935391 2.6098193559677427, 0. 3., 0.0384294391935391 3.390180644032257, 0.1522409349774265 3.7653668647301792, 0.3370607753949091 4.111140466039204, 0.5857864376269046 4.414213562373095, 0.8888595339607956 4.662939224605091, 1.2346331352698194 4.847759065022573, 1.6098193559677427 4.961570560806461, 2. 5., 2.390180644032257 4.961570560806461, 2.76536686473018 4.847759065022573, 3.1111404660392035 4.662939224605091, 3.414213562373095 4.414213562373096, 3.6629392246050907 4.111140466039204, 3.847759065022573 3.765366864730181, 3.961570560806461 3.3901806440322573, 4. 3.))
+        };
+        // This one is pretty close... but not quite right. Maybe add a parameter? It should be the same as the point buffer though?
```

Comment:
Stale comment?

Reply:
Oh, good catch! This actually represents a badly labeled TODO. I'm going to convert this to a draft.